src/python/pants/engine/legacy/query.py

```diff
@@ -0,0 +1,143 @@
+# coding=utf-8
+# Copyright 2019 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+from abc import abstractmethod
+
+from future.utils import text_type
+
+from pants.engine.legacy.graph import OwnersRequest
+from pants.engine.rules import RootRule, rule
+from pants.engine.selectors import Select
+from pants.scm.subsystems.changed import ChangedRequest, IncludeDependees
+from pants.util.meta import AbstractClass, classproperty
+from pants.util.objects import Exactly, TypedCollection, datatype
+from pants.util.strutil import safe_shlex_split
+
+
+class QueryComponent(AbstractClass):
```

Comment:
It seems strange that this would be in a package named `legacy`, as it's new functionality?

Reply:
Correct! This was because I thought `graph.py` had to import it and I was worried there'd be a cycle. I do not believe this is the case now, so this can be moved up to `src/python/pants/engine/`.

Reply:
Done!